src/hal/AngleSensorProto.cpp

```diff
@@ -1,4 +1,7 @@
 #include "AngleSensorProto.h"
+#include <algorithm>
+
+namespace Hal {
 
 const double AngleSensorProto::kMinAngle = 0.0;
 const double AngleSensorProto::kMaxAngle = 360.0;
```

Comment:
There's two alternatives here:

1. Add these constants only for the definition (.cpp), remove the members and make it avalable just here with an anonymous namespace (outside of Hal namespace).
2. If it's important to the class, initialize in the constructor but keeping the constants in an anonymous namespaces as mentioned before.
```cpp
namespace {
const double kMinAngleForSensor = 0.0;
const double kMaxAngleForSensor = 360.0;
}

AngleSensorProto::AngleSensorProto(double initialAngle)
    : angle_(initialAngle) 
    , kMinAngle(kMinAngleForSensor)
    , kMaxAngle(kMaxAngleForSensor)
{
}

```

Reply:
Instinctively I would say that those are inherent characteristics of the sensor, so I would go for the second approach.